parsers/ELEXON.py

```diff
@@ -84,17 +88,111 @@
+ZONEKEY_TO_INTERCONNECTOR = {
+    "BE->GB": ["Belgium (Nemolink)"],
+    "DK-DK1->GB": ["Denmark (Viking link)"],
+    "FR->GB": ["Eleclink (INTELEC)", "France(IFA)", "IFA2 (INTIFA2)"],
+    "GB->GB-NIR": ["Northern Ireland(Moyle)"],
+    "GB->IE": ["Ireland(East-West)"],
+    "GB->NL": ["Netherlands(BritNed)"],
+    "GB->NO-NO2": ["North Sea Link (INTNSL)"],
 }
 
 
+def query_elexon(url: str, session: Session, params: dict) -> list:
+    r: Response = session.get(url, params=params)
+    if not r.ok:
+        raise ParserException(
+            parser="ELEXON",
+            message=f"Error fetching data: {r.status_code} {r.reason}",
+        )
+    return r.json()
+
+
+def parse_datetime(settlementDate: str, settlementPeriod: int) -> datetime:
+    parsed_datetime = datetime.strptime(settlementDate, "%Y-%m-%d")
+    parsed_datetime += timedelta(hours=(settlementPeriod - 1) / 2)
+    return parsed_datetime.replace(tzinfo=timezone.utc)
+
+
+def query_production(
+    session: Session, target_datetime: datetime, logger: Logger
+) -> list:
+    """Fetches production data from the B1620 endpoint from the ELEXON API."""
+    production_params = {
+        "publishDateTimeFrom": (target_datetime - timedelta(days=2)).strftime(
+            "%Y-%m-%d 00:00"
+        ),
+        "publishDateTimeTo": target_datetime.strftime("%Y-%m-%d %H:%M"),
+    }
+    production_data = query_elexon(
+        ELEXON_URLS["production"], session, production_params
+    )
+    breakpoint()
+    parsed_events = parse_production(production_data, logger, "B1620")
+    return parsed_events
+
+
+def parse_production(
+    production_data: list[dict[str, any]], logger: Logger, dataset: str
+) -> ProductionBreakdownList:
+    """Parses production events from the ELEXON API. This function is used for the B1620 data or the FUELHH data."""
+    dataset_info = {
+        "B1620": {
+            "mode_mapping": RESOURCE_TYPE_TO_FUEL,
+            "mode_key": "psrType",
+            "quantity_key": "quantity",
+        },
+        "FUELHH": {
+            "mode_mapping": FUEL_INST_MAPPING,
+            "mode_key": "fuelType",
+            "quantity_key": "generation",
+        },
+    }
+
+    mode_mapping = dataset_info[dataset]["mode_mapping"]
+    mode_key = dataset_info[dataset]["mode_key"]
+    quantity_key = dataset_info[dataset]["quantity_key"]
+
+    all_production_breakdowns: list[ProductionBreakdownList] = []
+
+    for event in production_data:
+        production_breakdown = ProductionBreakdownList(logger=logger)
+        event_datetime = parse_datetime(
+            event.get("settlementDate"), event.get("settlementPeriod")
+        )
+        production_mix = ProductionMix()
+        storage_mix = StorageMix()
+
+        production_mode = mode_mapping[event.get(mode_key)]
+
+        if production_mode == "exchange":
+            continue
+
+        if production_mode == "hydro storage":
+            storage_mix.add_value("hydro", -1*event.get(quantity_key))
```

Comment:
This also needs to be casted to float and asserted that it's not none. Might make sense to create a helper function for that since we use it more than once.